src/Fantomas/AstTransformer.fs

```diff
@@ -62,1635 +34,1170 @@ module private Ast =
+            : TriviaNodeAssigner list =
+            match synExpr with
+            | SynExpr.Paren (expr, _, _, range) ->
+                visit
+                    expr
+                    (fun nodes ->
+                        [ mkNode SynExpr_Paren range
+                          yield! nodes ]
+                        |> finalContinuation)
+            | SynExpr.Quote (operator, _, quotedSynExpr, _, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
+                    [ visit operator; visit quotedSynExpr ]
+
+                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
+                    [ mkNode SynExpr_Quote range
+                      yield! (List.collect id nodes) ]
+                    |> finalContinuation
+
+                Continuation.sequence continuations finalContinuation
+            | SynExpr.Const (constant, range) ->
+                visitSynConst range constant
+                |> List.singleton
+                |> finalContinuation
+            | SynExpr.Typed (expr, typeName, _) ->
+                visit expr (fun nodes -> nodes @ visitSynType typeName |> finalContinuation)
+            | SynExpr.Tuple (_, exprs, _, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
+                    exprs |> List.map visit
+
+                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
+                    [ mkNode SynExpr_Tuple range
+                      yield! (List.collect id nodes) ]
+                    |> finalContinuation
+
+                Continuation.sequence continuations finalContinuation
+            | SynExpr.ArrayOrList (_, exprs, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
+                    exprs |> List.map visit
+
+                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
+                    [ mkNode SynExpr_ArrayOrList range
+                      yield! (List.collect id nodes) ]
+                    |> finalContinuation
+
+                Continuation.sequence continuations finalContinuation
+            | SynExpr.Record (_, _, recordFields, range) ->
+                mkNode SynExpr_Record range
+                :: (List.collect visitRecordField recordFields)
+                |> finalContinuation
+            | SynExpr.AnonRecd (_, _, recordFields, range) ->
+                mkNode SynExpr_AnonRecd range
+                :: (List.collect visitAnonRecordField recordFields)
+                |> finalContinuation
+            | SynExpr.New (_, typeName, expr, range) ->
+                visit
+                    expr
+                    (fun nodes ->
```

Comment:
I've got a post coming out soon where I talk about continuation-passing style all over again, and in the course of that, I ended up much preferring the following layout just on aesthetic grounds:

```fsharp
(fun nodes ->
    blah
)
|> visit expr
```
Saves a layer of indentation, although it is *quite weird* from an imperative-programming point of view.